velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -700,6 +711,23 @@ void SubstraitVeloxPlanConverter::constructFunctionMap(
   }
 }
 
+bool SubstraitVeloxPlanConverter::checkTypeExtension(
+    const ::substrait::Plan& substraitPlan) {
+  for (const auto& sExtension : substraitPlan.extensions()) {
+    if (!sExtension.has_extension_type()) {
+      continue;
+    }
+
+    // Only support UNKNOWN type in UserDefined type extension.
+    if (sExtension.extension_type().name() == "UNKNOWN") {
```

Comment:
this can be simplified to `return sExtension.extension_type().name() == "UNKNOWN";`

Reply:
No, the latest code as below. If there have type extension, it will then check whether the name  is unknown or not. If not, it will return false. If yes, it will do loop again. It will only return true if we have no typeExtension or typeExtension only have one extensionType with name unknown.
```
bool SubstraitVeloxPlanConverter::checkTypeExtension(
    const ::substrait::Plan& substraitPlan) {
  for (const auto& sExtension : substraitPlan.extensions()) {
    if (!sExtension.has_extension_type()) {
      continue;
    }

    // Only support UNKNOWN type in UserDefined type extension.
    if (sExtension.extension_type().name() != "UNKNOWN") {
      return false;
    }
  }
  return true;
}
```